coderd/database/queries/provisionerjobs.sql

```diff
@@ -50,42 +50,67 @@ WHERE
 	id = ANY(@ids :: uuid [ ]);
 
 -- name: GetProvisionerJobsByIDsWithQueuePosition :many
-WITH pending_jobs AS (
-    SELECT
-        id, created_at
-    FROM
-        provisioner_jobs
-    WHERE
-        started_at IS NULL
-    AND
-        canceled_at IS NULL
-    AND
-        completed_at IS NULL
-    AND
-        error IS NULL
+WITH filtered_provisioner_jobs AS (
+	-- Step 1: Filter provisioner_jobs and assign an order from upstream system
+	SELECT
+		*,
+		ROW_NUMBER() OVER () AS ordinality -- Track original order
```

Comment:
Is `ROW_NUMBER()` always the correct rank index? Should they be ordered by `created_at`?